### ISSUE TITLE
Conflict `Lint/AmbiguousRange` and `Style/RedundantParentheses` on complex literals (i.e. string interpolation)

### ISSUE THREAD

**Issue by Tietew** [rubocop/rubocop#10230]
When complex literal (i.e. string interpolation) in range literals (`..` and `...`), rubocop detects `Lint/AmbiguousRange`.
But adding parentheses causes `Style/RedundantParentheses`.

`Lint/AmbiguousRange` seems to be a false positive.
``` ruby
# dstr
"#{foo}-#{bar}"..'123-4567'
# xstr
`date`..'foobar'
# dsym
:"#{foo}-#{bar}"..:baz
# regexp
/a/../b/
```

`Lint/AmbiguousRange` seems to be reasonable, `Style/RedundantParentheses` should not be detected.
``` ruby
# array
[1]..[2]
# hash
{1 => 2}..{3 => 4}
```

--------

## Expected behavior

No offences should be detected.

## Actual behavior

Lint/AmbifuousRange detected.

```
"#{foo}-#{bar}"..'123-4567'
^^^^^^^^^^^^^^^ [Correctable] Lint/AmbiguousRange: Wrap complex range boundaries with parentheses to avoid ambiguity.
("#{foo}-#{bar}")..'123-4567'
^^^^^^^^^^^^^^^^^ [Correctable] Style/RedundantParentheses: Don't use parentheses around a literal.
```

## RuboCop version

```
$ be rubocop -V
1.22.3 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.7.4 x86_64-linux)
  - rubocop-performance 1.12.0
  - rubocop-rails 2.12.4
  - rubocop-rspec 2.5.0
```
